File: Python3/Two Sum.py
# 32ms, 100.0%
class Solution:
    def twoSum(self, nums, target):
        d = {}
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            else:
                d[target - nums[i]] = i
                

File: Java/Two Sum.java
// 1ms, 99.89%; 41.7MB, 5.65%
class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> rec = new HashMap<Integer, Integer>();
        int[] index = { 0, 0 };
        for (int i = 0; i < nums.length; ++i) {
            if (rec.containsKey(nums[i])) {
                index[0] = rec.get(nums[i]);
                index[1] = i;
                break;
            }
            rec.put(target - nums[i], i);
        }
        return index;
    }
}
